packages/react-app/src/App.jsx

```diff
@@ -44,32 +70,42 @@ function App({ mainnet, localProvider, appChainId }) {
   const displayToast = () => {
     setShowToast(true)
   }
+  console.log({ appState })
   let contractRef
+  const chainId = appState.chainId
   if (
-    externalContracts[selectedChainId] &&
-    externalContracts[selectedChainId].contracts &&
-    externalContracts[selectedChainId].contracts.REMIX_REWARD
+    externalContracts[chainId] &&
+    externalContracts[chainId].contracts &&
+    externalContracts[chainId].contracts.REMIX_REWARD
   ) {
-    contractRef = externalContracts[selectedChainId].contracts.REMIX_REWARD
+    contractRef = externalContracts[chainId].contracts.REMIX_REWARD
+  } else {
+    console.log('kosi externalContract')
   }
 
   useEffect(() => {
     async function getAddress() {
-      if (userSigner) {
-        const newAddress = await userSigner.getAddress()
-        // @ts-ignore
-        setConnectedAddress(newAddress)
-        console.log({ newAddress, userSigner })
+      const holderForConnectedAddress = await appState.provider.listAccounts()
+      console.log({ holderForConnectedAddress })
+      if (holderForConnectedAddress.length > 1 && connectedAddress) {
+        setConnectedAddress(holderForConnectedAddress[0])
       }
     }
     getAddress()
-  }, [userSigner])
+  }, [appState.provider, connectedAddress])
+
+  useEffect(() => {
+    appState.provider.on('chainChanged', chainId => {
```

Comment:
here you have to dispatch the right event depending on the `chainId`

Reply:
You are right. This is an error on my part. Fixing right away